python/lsst/ts/genericcamera/genericcameracsc.py

```diff
@@ -455,23 +570,162 @@ async def liveView_loop(self, exposure_time):
         self.isLive = False
         self.log.info("liveView_loop - End")
 
+    async def run_auto_exposure_loop(self, min_exp_time, max_exp_time, configuration):
+        """Run the auto exposure capture loop.
+
+        The cadence of the images is determined by the value of
+        AUTO_EXP_IMAGE_INTERVAL.
+
+        Parameters
+        ----------
+        min_exp_time: `float`
+            The minimum exposure time to use.
+        max_exp_time: `float`
+            The maximum exposure time to use.
+        configuration: `str`
+            A Yaml string containing additional configuration
+            parameters.
+        """
+        self.log.info("autoExposure_loop - Start")
+        self.isAutoExposure = True
+
+        shutter = False
+        sensors = ""
+        keyValueMap = ""
+        obsNote = ""
+        if configuration != "":
```

Comment:
Consider doing this in the `do` method instead, and pass in a configuration dictionary instead. maybe using `kwargs`?

Reply:
Done passing a dict.